engine.py

```diff
@@ -116,26 +116,12 @@ def post_app_init(self):
         """
         Do any initialization after apps have been loaded
         """
-        if self.has_ui:
-            # create the Shotgun menu
-            tk_mari = self.import_module("tk_mari")
-            self._menu_generator = tk_mari.MenuGenerator(self)
-            self._menu_generator.create_menu()
+        self.create_menu()
 
-        # Update the current Mari project with the current context so that next time this project is
-        # opened, the work area changes accordingly.
-        #
-        # Note, currently this will happen every time as long as Shotgun is running so it will update
-        # any opened projects even if they were never previously opened in Shotgun...  This just adds
-        # some metadata to the project though so shouldn't be a big deal!  If it is then we should
-        # make it less automatic.
-        current_project = mari.projects.current()
-        if current_project:
-            self.log_debug("Updating the Work Area on the current project to '%s'" % self.context)
-            self.__metadata_mgr.set_project_metadata(current_project, self.context)
```

Comment:
I believe the metadata setting should be handled by projectmanager (or workfiles, if that is enabled)
So exposed a public method to set_metadata and removed this from post_app_init and post_context_change.